url21b471891f_Agetac_Model_tgz-pJ8-org_json_TestJ8/src/org/agetac/common/model/impl/Vehicule.java

```diff
@@ -0,0 +1,171 @@
+	
+	private static final List<EtatVehicule> VALUES_EtatVehicule = Collections.unmodifiableList(Arrays.asList(EtatVehicule.values()));
+	private static final int SIZE_EtatVehicule = VALUES_EtatVehicule.size();
+	
+	private String caserneName;
+	private EtatVehicule etat;
+	private String groupeID;
+	private HashMap<EtatVehicule, String> groupesHoraires; 
+	private CategorieVehicule categorie;
+	
+	public Vehicule() {
+		super(null, null, new Position(0, 0));
+	}
+	
+	public Vehicule(String uid, Position position, CategorieVehicule cat, String caserneName, EtatVehicule etat, Groupe groupe, String heure) {
+		super(uid, cat.name()+" "+caserneName, position);
+		this.categorie = cat;
+		this.caserneName = caserneName;
+		this.etat = etat;
+		this.groupeID = groupe.getUniqueID();
+		this.groupesHoraires = new HashMap<EtatVehicule, String>();
+		this.groupesHoraires.put(EtatVehicule.DISPO_CASERNE, heure);
+		this.groupesHoraires.put(EtatVehicule.ALERTE, heure);
+		this.groupesHoraires.put(EtatVehicule.PARTIS, heure);
+		this.groupesHoraires.put(EtatVehicule.SUR_LES_LIEUX, "");
+		this.groupesHoraires.put(EtatVehicule.TRANSPORT_HOPITAL, "");
+		this.groupesHoraires.put(EtatVehicule.DISPO_RADIO, "");
+		this.groupesHoraires.put(EtatVehicule.TEMPS_DEPASSE, "");
+		this.groupesHoraires.put(EtatVehicule.DEMOBILISE, "");
+	}
+	
+	public Vehicule(String uid, Position position, CategorieVehicule cat, String caserneName, EtatVehicule etat, String groupeUID, String heure) {
+		super(uid, cat.name()+" "+caserneName, position);
+		this.categorie = cat;
+		this.caserneName = caserneName;
+		this.etat = etat;
+		this.groupeID = groupeUID;
+		this.groupesHoraires = new HashMap<EtatVehicule, String>();
+		this.groupesHoraires.put(EtatVehicule.DISPO_CASERNE, heure);
+		this.groupesHoraires.put(EtatVehicule.ALERTE, heure);
+		this.groupesHoraires.put(EtatVehicule.PARTIS, heure);
+		this.groupesHoraires.put(EtatVehicule.SUR_LES_LIEUX, "");
+		this.groupesHoraires.put(EtatVehicule.TRANSPORT_HOPITAL, "");
+		this.groupesHoraires.put(EtatVehicule.DISPO_RADIO, "");
+		this.groupesHoraires.put(EtatVehicule.TEMPS_DEPASSE, "");
+		this.groupesHoraires.put(EtatVehicule.DEMOBILISE, "");
+	}
+	
+
+	
+	public Vehicule(String uid, String nom, Position position, String caserneName, EtatVehicule etat, String groupeID) {
+		super(uid, nom, position);
+		this.caserneName = caserneName;
+		this.etat = etat;
+		this.groupeID = groupeID;
+		this.groupesHoraires = new HashMap<EtatVehicule, String>();
+	}
+	
+	public Vehicule(JSONObject json) throws InvalidJSONException {
+		super(json);
+		try {
+			this.categorie = CategorieVehicule.valueOf(json.getString("categorie"));
+			this.etat = EtatVehicule.valueOf(json.getString("etat"));
+			this.groupeID = json.getString("groupeID");
+			this.caserneName = json.getString("caserneName");
+			//TODO: ComplĂŠter avec la liste des groupe horaire
+		} catch(JSONException e){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)